### DISCUSSION TITLE
MA2: Choose to play complete folder when browsing a folder structure (ie NAS)

### DISCUSSION THREAD

**Discussion by rvandet** [music-assistant/support#1256] [Feature requests and ideas]
As a User i wish to start playing all items in a folder
So that i can browse my music library, i can play all items with on click only.

In MA it is possible, but in the way i do it, it requires many clicks:
- choose a folder
- choose a track
- list albums
- play album

When the album (tags) is different from the content of the folder, this does not work properly.

Below is a screen shot from MPD player, an app in the Apple and Android playstores. Playing a folder is done via the tree points on the right of the folder. That is very intuitive.

![Afbeelding van WhatsApp op 2023-05-21 om 12 02 14](https://github.com/music-assistant/hass-music-assistant/assets/11903294/af308036-118d-4471-b3e2-18757ccee81e)

**Comment by OzGav** [comment]
Based on discussions on GitHub I think this is unlikely to get up. MA is not designed to play music from a folder. If you want that there is probably other simpler solutions. If you only want to use MA then your options are to either (a) add these individual tracks to a playlist or (b) tag them with a made up album name and then you can select the album to play.

To quote Marcel "Music Assistant is not about playing FILES but indexing and playing artists, albums and tracks."

**Reply by marcelveldt** [reply]
Correct but let's leave this as feature request to either implement on a rainy day or when there's high user demand for such a feature. For now, the focus is not really at browsing folders and files in MA but using the library structure of Artists, Albums, Tracks and Playlists

**Comment by rvandet** [comment]
Fair enough, with this in mind my suggestion would be:

When browsing a folder, it would make sense that the link to the Album (derived from the Metadata) can be a big bigger so that it is easier ti navigate to the Album.

**Reply by OzGav** [reply]
Whilst MA will create the library based on the metadata it is also recommended to follow the file structure shown in the example here https://kodi.wiki/view/Music_tagging#Folder_and_File_Structure The usage intent of MA is that you are NOT using the browse function to peruse your music. The intent is that you are searching through the artists, albums and tracks. Hence the order shown in the interface. For those with multiple music providers it makes no sense to use the browse option as MA creates one large library of all available music. The browse option is there to all selective adding of items to the MA library (e.g radio stations via radio browser).

Again if you just want to browse through your folders then the HA media functionality may be more suitable.

**Comment by GreenhouseAU** [comment]
I would absolutely love to have this feature.

My music is sorted chronologically by folder name, eg:

2024.04.05 Only God Was Above Us
2024.05.03  Reasonable Woman
2024.05.31  Secrets Of The Future
2024.08.09  Sunday Sadness
2024.09.06  The Second Act

It's still tagged normally, but would be great to have the option to just play a folder. Looking for album art in that folder would be amazing (folder.jpg) :⁠-⁠)

**Comment by Swimmer71** [comment]
I also would love to play from folder, stored on my syno NAS. My tracks are tagged within the folder with artist, titel and genre. Unfortunately genre is not used by MA and there is no way to simple select a complete folder to enque to playlist.

Even I did not manage to create a playlist out of a folder. The conext menu (...) of folders do not show menu entries. How can I create a playlist with all songs of a folder on NAS?

The only way now for me is to tag all tracks in folder with albumname and/or albumartist with foldername to see them in album or artist.

**Reply by OzGav** [reply]
You can use something like VLC to create the playlist

**Reply by Swimmer71** [reply]
I did so, but Hits-neu.m3u shows no tracks in MA (here screenshot of the first entries).
I also tried to use hostname NAS instead of IP.
![Screenshot 2024 11 19-10 49](https://github.com/user-attachments/assets/87cba099-1ba3-49dc-a1f3-3d654855406c)

**Reply by OzGav** [reply]
https://music-assistant.io/faq/how-to/#create-playlists-or-use-m3u-files

**Comment by Swimmer71** [comment]
Or alternatively: how can I access to the DLNA server of my synology. Generally I don't see DLNA....

**Comment by lionking102** [comment]
I would also love support for this.
Sometimes folders are just the best way to browse music for me. Sadly nearly all music-streaming servers for selfhosting don`t support them.
I love music-assistant and would be really happy to get a bit better support for folders. Playing all files in a folder and maybe also album covers for the folder.
Besides that: Thanks a lot for the awesome ongoing work in music assistant!

**Comment by OzGav** [comment]
Playing a folder from the browse view is available in 2.4.0b28

**Comment by Swimmer71** [comment]
This is great, thanks for that feature. How can I select a folder now via action music_assistant.play_media?
Under media_type there is no option for folder

![Screenshot 20250215-125210](https://github.com/user-attachments/assets/156bffbe-bbb9-4b05-bac7-6b31011489e3)

**Reply by OzGav** [reply]
That is a separate feature request that needs to be raised.

**Reply by marcelveldt** [reply]
use filesystem://folder/relative/path/to/folder as media id

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
